### ISSUE TITLE
Rover suspenssion system 

### ISSUE THREAD

**Issue by Vaslouk** [vasilepi/Space-Rover#18]
For visual and also to further understand any not obvious problems references mostly I wanted to make a 3D early draft representation of the rover suspension system again this not the final design, however it does comply with the regulations we have settled.I will also have attached some screenshots so opening inventor is not nessesary.
![image](https://github.com/vasilepi/Space-Rover/assets/149177726/b3634145-82b2-4a27-afed-d274dc256371)

**Comment by vasilepi** [comment]
[This](https://github.com/vasilepi/Space-Rover/blob/main/Mobility/att.drawio) drawio file is for the fbds. You can use it as a reference and we can work together on this

**Comment by vasilepi** [comment]
There is also another issue for this. I will close that and we can work via this one

**Comment by Vaslouk** [comment]
> For visual and also to further understand any not obvious problems references mostly I wanted to make a 3D early draft representation of the rover suspension system again this not the final design, however it does comply with the regulations we have settled.I will also have attached some screenshots so opening inventor is not nessesary. ![image](https://private-user-images.githubusercontent.com/149177726/285992311-b3634145-82b2-4a27-afed-d274dc256371.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEyNjg2NzgsIm5iZiI6MTcwMTI2ODM3OCwicGF0aCI6Ii8xNDkxNzc3MjYvMjg1OTkyMzExLWIzNjM0MTQ1LTgyYjItNGEyNy1hZmVkLWQyNzRkYzI1NjM3MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTEyOVQxNDMyNThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNmFiMGEzMDNiNGFlOWE1NDlhYTU2ODcwMGQ4MDQxOTZhODc5YmE3MDAzOGEyODFjNDA2ZTQ3NTgwY2M5NTllJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.GfYWr2j_sIUGMozUuS5Jh_f6QDCvq9hghDoxbgC_4Oo)

After some consideration I decided to redisign the parts of the suspenssion system focusing on the desing of the joints and also regarding the effectiveness of thier shape and thier weight ratio so they can be as litte as possoble but also strong enough to withstand the forces that are being delt with.I will attach some screenshots so you can take a look without having to open inventor.In adittion to that I will attach the propssed desing of the joint.
![20231129_164247](https://github.com/vasilepi/Space-Rover/assets/149177726/2456c776-dfbb-4884-b8af-7f15d8aade5d)
![rovers](https://github.com/vasilepi/Space-Rover/assets/149177726/98466c96-56c5-4930-affc-13011191056e)

**Comment by Vaslouk** [comment]
I wanted to clarify the design of the joints and to ask for your comments thus I am attaching a more detailed draft of them
![Uploading 20231130_105112.jpg…]()

**Comment by vasilepi** [comment]
> I wanted to clarify the design of the joints and to ask for your comments thus I am attaching a more detailed draft of them
Uploading 20231130_105112.jpg…

What are these joints for? I see you have added two bearings on a shaft. Can you elaborate further. Also I am not really sure what each of the CAD components are

**Comment by Vaslouk** [comment]
![20231130_105112](https://github.com/vasilepi/Space-Rover/assets/149177726/87a44fcb-d905-4482-aec6-70efd766081f)

**Comment by Vaslouk** [comment]
> > I wanted to clarify the design of the joints and to ask for your comments thus I am attaching a more detailed draft of them
> > Uploading 20231130_105112.jpg…
> 
> What are these joints for? I see you have added two bearings on a shaft. Can you elaborate further. Also I am not really sure what each of the CAD components are

the joints are the parts that keep the rods in place but lets them rotate

**Comment by vasilepi** [comment]
>the joints are the parts that keep the rods in place but lets them rotate

So the hub I see is supposed to be the horizontal rod able to rotate

**Comment by Vaslouk** [comment]
Exactly

**Comment by Vaslouk** [comment]
Exactly

**Comment by vasilepi** [comment]
Based on your draft I designed a more representative model of the mobility system. I redesigned the revolution joint so that we don't need two parts for the horizontal rod. We can move on and calculate these parts with dimension I used and then refine them if necessary. @Vaslouk 
| description | pic |
|------------|----|
|assembly 1 | ![image](https://github.com/vasilepi/Space-Rover/assets/132815921/265acb06-05af-4aaa-8eac-01405f276f62) | 
| assembly 2 | ![image](https://github.com/vasilepi/Space-Rover/assets/132815921/5693780c-20c6-4d5d-baa8-29554753f5dc)|
| horizontal rod | ![image](https://github.com/vasilepi/Space-Rover/assets/132815921/6122e00c-b452-437a-b4ac-1b7451f17171)|
|vertical rod | ![image](https://github.com/vasilepi/Space-Rover/assets/132815921/6922d4cf-dec7-403a-9c7a-06fea557047b) |
| arced rod | ![image](https://github.com/vasilepi/Space-Rover/assets/132815921/dfd0dc13-369a-4c32-8ab7-1b029c6a9926) |
|revolution joint | ![image](https://github.com/vasilepi/Space-Rover/assets/132815921/a692b949-bb78-4ccd-8835-c2f48af4921b)|

**Comment by Vaslouk** [comment]
Although I like the changes you have made the revolution joint doesn't work because the horizontal rod which rotates is in contact with the shaft lock.

**Comment by vasilepi** [comment]
> Although I like the changes you have made the revolution joint doesn't work because the horizontal rod which rotates is in contact with the shaft lock.

You are right that's my mistake. The only change then should be to avoid the contact of these two parts and let the lock touch the interior ring of the bearing

**Comment by vasilepi** [comment]
> > Although I like the changes you have made the revolution joint doesn't work because the horizontal rod which rotates is in contact with the shaft lock.
> 
> You are right that's my mistake. The only change then should be to avoid the contact of these two parts and let the lock touch the interior ring of the bearing

Correction:
![image](https://github.com/vasilepi/Space-Rover/assets/132815921/46e98270-b41d-4b96-a77f-829759d56d8e)
I will upload the step files for you

**Comment by Vaslouk** [comment]
Actully we can easily fix this problem by adding one axial bearing on each side, I should also add that this idea was thought initially by @kotidius.

**Comment by Vaslouk** [comment]
This what I am talking about 
![20231206_103940](https://github.com/vasilepi/Space-Rover/assets/149177726/36db98fc-6760-41af-bba4-5a8981414754)

**Comment by Vaslouk** [comment]
![image](https://github.com/vasilepi/Space-Rover/assets/149177726/6caa35fc-9b62-4ef7-8120-8d3b9c9520d2)

**Comment by Vaslouk** [comment]
![image](https://github.com/vasilepi/Space-Rover/assets/149177726/3f785437-b715-4247-b81a-fecc38891b6b)

**Comment by vasilepi** [comment]
# SAFETY FACTORS FOR THE MOBILITY SYSTEM IN STATIC LOADING
## 1. Design
The mobility system looks like this 
![image](https://github.com/vasilepi/Space-Rover/assets/132815921/b0ca066a-d5a3-43ed-b084-ca25884c0762)

## 2. Problem analyze
The parts that need to be calculated are:
- vertical bar (cyan)
- horizontal bar (purple)
- arced beam (orange)

Analysis will be done for:
- **maximum slope of $30^\circ$ angle**
- **impact**

Safety factors will be calculated as:

$$
SF=\frac{s_y}{|\sigma_{applied}|}
$$

Where $s_y$ will the the yield strength of the material

## 3. Solve
### 3.1 FBD
The FBD for the system looks like this:
![image](https://github.com/vasilepi/Space-Rover/assets/132815921/71731f68-71e1-4bb4-a9d9-629a5c0cf715)
Where $f$ would be the slope angle!
In order to calculate the safety factors, the forces need to be calculated first. A first order linear system of 12 equations with 12 unknowns is created:
$$\boldsymbol{A}x=\boldsymbol{Z}$$
For all the bars it is:
$$\sum F_x=0$$
$$\sum F_y=0$$
$$\sum M=0$$
where x would be the axis in parallel to the bar's center line and y would be the perpendicular to the bar's center line axis.\
For both the bar's the moment equilibrium is taken at their left hand point. (11 and 21 correspondingly). In total there is 3 equations for each horizontal bar. So 6 equations.\
Forces equilibrium produces another two equations for each of the vertical bars as well. In total 4 equations from these bars.\
For the last two equations, the mobility system is checked at a high level where only the weight forces as well as the ground normals and friction act. And thus forces equilibrium on each axis of the high level system produces another two equations.
Here:


| variable | Description |
| --------| ------------|
| $s$ | $sinf$ |
| $c$ | $cosf$ |
| $l$ | horizontal bar length |
| $w$ | $W/8$ |

$$\boldsymbol{A}=\begin{bmatrix}
-s & -s & 0 & 0 & c & c & 0 & 0 & 0 & 0 & 0 & 0\\
c & c & 0 & 0 & s & s & 0 & 0 & 0 & 0 & 0 & 0\\
0 & -sl & 0 & 0 & 0 & cl & 0 &  0 & 0 & 0 & 0 & 0\\
0 & 0 & -s & -s &  0 &  0 & c &  c & 0 & 0 & 0 & 0\\
0   & 0 & c  &  c & 0  & 0 & s   & s  & 0  & 0 & 0 & 0\\
0  &  0 & 0 & -sl & 0 &  0 & 0 & cl & 0 & 0 & 0 & 0\\
0   & 0 & 0  &  0 & -1  & 0 & -1  & 0 & c & 0 & s & 0\\
-1   & 0 & -1 &   0 & 0 & 0 & 0 &  0 & -s & 0 & c & 0\\
 0   & 0 & 0   & 0 & 0  & 0 & 0  & 0 & 1 & 1 & 0 & 0\\
 0   & 0 & 0 &   0 & 0 & -1 & 0 & -1 & 0 & c & 0 & s\\
 0  & -1 & 0   & -1 & 0  & 0 & 0  & 0 & 0 & -s & 0 & c\\
 0   & 0 & 0 &   0 & 0 &  0 &  0 &  0 & 0 & 0 & 1 & 1\\
\end{bmatrix},\ \boldsymbol{Z}=\begin{bmatrix}
cw\\ 
ws\\ 
wc \frac{l}{2} \\ 
wc \\ 
ws \\ 
wc\frac{l}{2} \\ 
0\\ 
0\\ 
2wc\\ 
0\\ 
0 \\ 
2ws\\
\end{bmatrix}$$

Of course:

$$x=\begin{bmatrix}
a_{11}\\
a_{12}\\
a_{21}\\
a_{22}\\
b_{11}\\
b_{12}\\
b_{21}\\
b_{22}\\
N_1\\
N_2\\
T_1\\
T_2
\end{bmatrix}$$



### 3.2 Horizontal Bar (purple one)
#### 3.2.1 Safety Factors
Safety factors for bending and tension are obtained using MNQ method. Considering a rectangular cross section with b: base, h: height the applied stress is calculated as:
$$\sigma_z^{hor}=\frac{N}{A_h}$$
$$\sigma_b^{hor}=\frac{N}{A_h} +\frac{M}{I_h}z_h$$
The critical cross section for the bending load would be in the middle of the bar, while for the tension it is calculated from MATLAB. 

#### 3.2.2 Deflection
The deflection across the vertical axis from a load at the center of the beam P, is calculated as:
$$\delta_y=\frac{P\cdot l^3}{48EI_h}$$
[source](https://www.omnicalculator.com/construction/beam-deflection#how-to-calculate-the-maximum-beam-deflection)

### 3.3 Vertical Bar (cyan)
The vertical bar is calculated at bending and at impact.
#### 3.3.1 Bending
For the bending the process stays the same:
| Vertical Bar | MNQ | Geometry |
|-----|-----|---|
| ![image](https://github.com/vasilepi/Space-Rover/assets/132815921/6d04b287-e2d7-4eef-8c04-5150b5bf15bc) | ![image](https://github.com/vasilepi/Space-Rover/assets/132815921/6cde81e3-a3a3-46d5-98f3-22685a88ff17) | ![image](https://github.com/vasilepi/Space-Rover/assets/132815921/dda8cb88-7f84-4bd6-8a58-1eb2d67a5888) |

$$
    a=180^\circ - qq^\circ
$$

$$
    \zeta=90^\circ - a^\circ
$$

$$
    x_{max}=\frac{B}{sin\zeta}
$$

$$
    N_{max}=-T\cdot c\cdot cos\zeta -(N\cdot c +T\cdot s)\cdot sin\zeta +N\cdot s \cdot cos\zeta
$$

$$
    M_{max}=-T\cdot c\cdot sin\zeta \cdot x_{max} +(N\cdot c +T\cdot s)\cdot cos\zeta \cdot x_{max} +N\cdot s \cdot sin\zeta \cdot x_{max}
$$

Finally:

$$\sigma_b^{ver}=\frac{N_{max}}{A_v} +\frac{M_{max}}{I_v}z_v$$

#### 3.3.2 Impact
[source](https://www.clear.rice.edu/mech403/HelpFiles/ImpactLoadFactors.pdf)
In the conceptual design stages in order to calculate for an impact you should approximate the impact dynamic load with a static one. This can be done with an _impact load factor_. For a mass moving horizontally with velocity of $V$ this is:
$$n=\sqrt{\frac{\eta V^2}{g\delta_{static}}}$$
Where $\eta$ is the efficiency that show how much of the kinetic energy is converted. An $\eta=1$ would mean the the body is elastic and all of the kinetic energy is converted. According to [datasheets](https://www.researchgate.net/figure/Restitution-coefficients-for-various-metallic-and-polymer-materials-and-their_tbl2_51153031) for specific material the efficiency can be found. Also $\delta_{static}$ would be the equivalent static deflection at the impact load's direction as if the impact load would be a static one.
The impact load would be the momentum: $$F=m_{rover}V$$
Calculating the deflection for a cantilever beam $\delta$ it would that:\
$$\delta_{static}=\delta \cdot cos\theta $$


![image](https://github.com/vasilepi/Space-Rover/assets/132815921/56a078d1-3545-4f64-85af-790b01de41a7) ![image](https://github.com/vasilepi/Space-Rover/assets/132815921/dfdb1d72-514c-43eb-8323-aa3ddc67800a)\

From Statics:

$$
   \delta=\frac{Fcosa\cdot x_{max}^3}{3EI_v}
$$

$$
   \theta=\frac{Fcosa\cdot x_{max}^2}{2EI_v}
$$

And also from Mechanics:

$$
   F_{se}=F \cdot n
$$


$$
   N_{max}=-F_{se}sina
$$

$$
   M_{max}=-F_{se}cosa\cdot x_{max}
$$

While the applied stress remains the same!

### 3.4 Arced Beam
#### 3.4.1 Safety Factor
For the safety factor in bending similar MNQ process is required. 

| Arced Beam | MNQ yx | MNQ zy |
|--|--|--|
| ![image](https://github.com/vasilepi/Space-Rover/assets/132815921/434196e4-77e8-460a-8113-8f0f296c4919) | ![image](https://github.com/vasilepi/Space-Rover/assets/132815921/91ea9d26-5e32-4b3b-81ae-476a034887b8)| ![image](https://github.com/vasilepi/Space-Rover/assets/132815921/dbeb183d-23b9-48c2-9636-035af0986aae) |

$$
   M_{yx}=N\cdot R\cdot sin\theta
$$

$$
   M_{zy}=T\cdot R\cdot (1-cos\theta)
$$

$$\theta_{max}=90^\circ$$

$$
  \sigma_{b}^{arc}=\frac{M_{zy}}{I_{zy}}z_{zy}+\frac{M_{yx}}{I_{yx}}z_{yx}
$$

 #### 3.4.2 Deflection
For the deflection it is:

$$
  \delta_y=\frac{\frac{\pi}{4}NR^3}{EI}
$$

[source](https://wiki.lbto.org/pub/FLAO/RoarksFormulasForStressAndStrain/2542x_09.pdf)

The deflection of the linear part of the beam should be about equal as the if the deflection were to happen in the arced part as:

<img src="https://github.com/vasilepi/Space-Rover/assets/132815921/b764e105-bec9-455d-8421-1422210f16d9" width="400" height="400">

## Results
According to #7 rover's weight is $2373.66\ N$ and velocity is $V=0.1\ m/s$. Also the material used was Aluminum 6061-T6 for all the parts.
Geometry:

|Characteristic | Value |
|-|-|
| $f$ | $30^\circ$ |
| $qq$ | $140^\circ$ |
| $b_h$ | $30\ mm$ |
| $h_h$ | $30\ mm$ |
| $b_v$ | $20\ mm$ |
| $h_v$ | $30\ mm$ |
| $b_a$ | $18\ mm$ |
| $h_a$ | $22\ mm$ |
| $B$ | $151.26\ mm$ |
| $l$ | $440\ mm$ |

The forces in (N):

$$x=    \begin{bmatrix}
0.0000\\
-0.0000\\
-0.0000\\
0.0000\\
148.3537\\
148.3537\\
148.3538\\
148.3537\\
256.9562\\
256.9562\\
148.3537\\
148.3537\\
\end{bmatrix}
$$

<img src="https://github.com/vasilepi/Space-Rover/assets/132815921/b3e33741-9271-4b0f-a48f-d73f465d5fb9" width="400" height="400">

The results to come from the MATLAB script are:

|Ch.|What|Value|
|-|-|-|
|_3.2.1_|$SF_z^{hor}$|$1.1163\times10^3$|
|_3.2.1_|$SF_b^{hor}$|$42.2768$|
|_3.2.2_|$\delta_{hor}$|$0.0979\ mm$|
|_3.3.1_|$SF_b^{ver}$|$22.6711$|
|_3.3.2_|$SF_i^{ver}$|$347.1974$|
|_3.4.1_|$SF_b^{arc}$|$10.3487$|
|_3.4.2_|$\delta_{arc}$|$0.6180\ mm$|

If needed further reducing of the geometrical values can happen due to the high values of the safety factors!
The CADs have been updated on OnShape. The ones to be used are ("horizontal rod", "steering system, "vertical redesign")

**Comment by vasilepi** [comment]
Just uploaded the parts [here](https://github.com/vasilepi/Space-Rover/tree/mobility_cads_iteration1/Mobility/CADs)
@Vaslouk

**Comment by Vaslouk** [comment]
![image](https://github.com/vasilepi/Space-Rover/assets/149177726/5a0368a3-fa6f-4bc1-9be1-4c95e4825ea9)
we will talk about this on the metting

**Comment by Vaslouk** [comment]
![image](https://github.com/vasilepi/Space-Rover/assets/149177726/e09fdd28-ac65-4731-a01b-673ef6441b74)
Final desing of turning joints

**Comment by Vaslouk** [comment]
![image](https://github.com/vasilepi/Space-Rover/assets/149177726/b31a6e11-b313-4f65-bd51-70a4cc90c863)
turning joint final desing
